packages/fxa-shared/db/models/auth/recovery-key.ts

```diff
@@ -64,16 +64,12 @@ export class RecoveryKey extends BaseAuthModel {
     }
   }
 
-  static async updateRecoveryKeyHint({
-    uid,
-    hint,
-  }: Pick<RecoveryKey, 'uid' | 'hint'>) {
+  static async updateRecoveryKeyHint(uid: string, hint: string) {
     try {
-      const uidBuffer = uuidTransformer.to(uid);
       const result = await RecoveryKey.query()
-        .where(uid, uidBuffer)
-        .update({ hint });
-      return !!result;
+        .where('uid', uuidTransformer.to(uid))
```

Comment:
Q: Any risk of `uuidTransformer.to(uid)` returning something unexpected and updating multiple records? (or maybe `uid` is a primary key and it'd be a 0-or-1 record scenario).

Reply:
`uid` is the primary key for the `recoveryKeys` table 👍